### ISSUE TITLE
Does jmespath support "substring" function

### ISSUE THREAD

**Issue by zacard-orc** [jmespath/jmespath.js#77]
hello, i met a problem ,to use different char idx to switch the business. eg

"01234abc" : extract if ab => way to 1
"01234efg" : extract if ef => way to 2
"01234mno" : extract if mn=> way to 3

I browse the official doc and not found any method to support this. 

Dose anybody met the same problem and how to resolve it ?

**Comment by darrenmothersele** [comment]
Hi, sorry - I don't understand your description of what you are trying to do. Can you give an example (or two) of expected input and expected output?

**Comment by jarrodek** [comment]
I have a similar issue. I would like to have a built-in function that performs a similar to `String.prototype.substring()` operation.

For given input:

```json
{
  "foo": "bar"
}
```

I would like to run an expression like this:

```js
// in JS the substring has a signature of substring(indexStart, indexEnd)
.search(..., 'foo | substring(0, 1)') // ba
.search(..., 'foo | substring(0, 2)') // bar
.search(..., 'foo | substring(0)') // bar
.search(..., 'foo | substring(1)') // ar
```
or by using slice expression

```js
.search(..., 'foo[0:1]') // ba
.search(..., 'foo[0:2]') // bar
.search(..., 'foo[0]') // bar
.search(..., 'foo[1]') // ar
```

**Comment by springcomp** [comment]
@zacard-orc we are standardizing a host of [string manipulation functions](https://github.com/jmespath-community/jmespath.spec/discussions/21#discussion-3914002) in JMESPath Community. Substrings will be achieved [using the existing `slice-expression` syntax](https://github.com/jmespath-community/jmespath.spec/discussions/26#discussioncomment-3284127).

**Comment by jpmckinney** [comment]
JEP: https://github.com/jmespath/jmespath.jep/issues/5

**Comment by springcomp** [comment]
> JEP: [jmespath/jmespath.jep#5](https://github.com/jmespath/jmespath.jep/issues/5)

Yep ! that’s [exactly where](https://github.com/jmespath-community/jmespath.spec/discussions/26) I took this piece of feedback when brainstorming the [list of features](https://github.com/jmespath-community/jmespath.spec/discussions/38) to include in the next version of JMESPath. Incidentally, @glenveegee has created an awesome [port of JMESPath to TypeScript](https://github.com/nanoporetech/jmespath-ts) but this does not seem to be maintained any longer.

So, while preparing [a version for JMESPath Community](https://github.com/jmespath-community/typescript-jmespath/pulls), I am proposing all my changes [upstream](https://github.com/nanoporetech/jmespath-ts/pulls/springcomp) anyway. I hope to find an official maintainer then.

**Comment by glenveegee** [comment]
@springcomp I'm updating it as we speak. Just been busy.
